VideoLocker/src/main/java/org/edx/mobile/view/CourseOutlineFragment.java

```diff
@@ -144,18 +144,35 @@ public void updateMessageView(View view){
     private void initializeAdapter(){
         if (adapter == null) {
             // creating adapter just once
-            adapter = new CourseOutlineAdapter(getActivity(), environment.getDatabase(), environment.getStorage(), new CourseOutlineAdapter.DownloadListener() {
+            adapter = new CourseOutlineAdapter(getActivity(), environment.getDatabase()
+                    , environment.getStorage(), new CourseOutlineAdapter.DownloadListener() {
                 @Override
                 public void download(List<HasDownloadEntry> models) {
-                    downloadManager.downloadVideos(
-                            (List) models, (FragmentActivity) getActivity(), (VideoDownloadHelper.DownloadManagerCallback) getActivity());
-
+                    if (AppConstants.offline_flag) {
+                        ((BaseFragmentActivity) getActivity()).
+                                showInfoMessage(getString(R.string.wifi_off_message));
+                    } else {
+                        downloadManager.downloadVideos(
+                                models, getActivity()
+                                , (VideoDownloadHelper.DownloadManagerCallback) getActivity());
```

Comment:
This can be condensed to two lines probably, and the comma should not start a line.